_posts/2025-11-18-signal-decision.md

```diff
@@ -0,0 +1,620 @@
+---  
+layout: post
+title: "Signal-Decision Driven Architecture: Reshaping Semantic Routing at Scale"
+author: "vLLM Semantic Router Team"
+image: /assets/logos/vllm-logo-text-light.png
+---
+
+The earlier versions of vLLM Semantic Router relied on classification-based routing, a straightforward approach where user queries are classified into one of 14 MMLU domain categories, and then routed to corresponding models. While this worked for basic scenarios, we quickly discovered its limitations when building production AI systems for enterprises.
+
+Consider this real-world scenario: A user asks, "I need urgent help reviewing a security vulnerability in my authentication code." The classification-based router would identify this as a "computer science" query and route it to a general coding model. But it misses critical context:
+
+- The **urgency** signal that requires immediate attention
+- The **security** sensitivity that demands specialized expertise and jailbreak protection
+- The **code review** intent that benefits from reasoning capabilities
+- The **authentication** complexity that needs careful analysis
+
+This single example reveals the fundamental constraint: **classification-based routing captures only one dimension of user intent—the domain—while ignoring the rich, multi-dimensional signals embedded in natural language queries.**
+
+Today, we're introducing the **Signal-Decision Architecture**—a complete reimagining of semantic routing that scales from 14 fixed categories to unlimited intelligent routing decisions. This new architecture combines multi-dimensional signal extraction, flexible decision logic with AND/OR operators, and built-in plugin orchestration to deliver production-ready semantic intelligence.
+
+![](/assets/figures/semantic-router/signal-0.png)
+
+## The Problem: Why Classification-Based Routing Doesn't Scale
+
+The previous vLLM Semantic Router architecture followed a simple pipeline:
+
+```text
+User Prompt → MMLU Domain Classification → Model Selection
+```
+
+This approach has several fundamental limitations that prevent it from scaling to enterprise requirements:
```

Comment:
Looks a little better in the rendered doc because we're using headings instead of a list

```suggestion
This approach has several fundamental limitations that prevent it from scaling to enterprise requirements.
```